geemap/core.py

```diff
@@ -402,13 +402,25 @@ class Map(ipyleaflet.Map, MapInterface):
         "scroll_wheel_zoom": True,
     }
 
-    _BASEMAP_ALIASES: Dict[str, str] = {
-        "DEFAULT": "OpenStreetMap.Mapnik",
-        "ROADMAP": "Esri.WorldStreetMap",
-        "SATELLITE": "Esri.WorldImagery",
-        "TERRAIN": "Esri.WorldTopoMap",
-        "HYBRID": "Esri.WorldImagery",
-    }
+    if common.google_maps_api_key() is not None:
+        _BASEMAP_ALIASES: Dict[str, str] = {
+            "OpenStreetMap": "OpenStreetMap.Mapnik",
+            "Google.Roadmap": "Google.Roadmap",
+            "Google.Satellite": "Google.Satellite",
+            "Google.Terrain": "Google.Terrain",
+            "Google.Hybrid": "Google.Hybrid",
+            "Google.Traffic": "Google.Traffic",
+            "Google.Streetview": "Google.Streetview",
```

Comment:
A couple suggestions for this code.
1) This is a backward-incompatible change. Users that previously set the basemap as "DEFAULT", "ROADMAP" or any of the other alises that existed will now get a different basemap.
2) We shouldn't have logic that branches if we're not in a function (and this is a constant).

Reply:
I have added the backward compability. The map alias should now work as usual. 

```python
import geemap.core as geemap
m = geemap.Map(basemap="HYBRID")
m
```